shared/health/src/main/java/io/pravega/shared/health/HealthServiceFactory.java

```diff
@@ -0,0 +1,61 @@
+/**
+ * Copyright Pravega Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.shared.health;
+
+import io.pravega.common.Exceptions;
+import lombok.NonNull;
+
+import java.time.Duration;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * Provides instances of a {@link HealthServiceManager} and optionally starts the {@link HealthServiceUpdater}.
+ */
+public class HealthServiceFactory implements AutoCloseable {
```

Comment:
I think we can further simplify this by removing this factory class. There is no added value in having a factory that creates a single type, which can be achieved by invoking that type's constructor.